src/coreclr/jit/vartype.h

```diff
@@ -354,6 +354,18 @@ inline bool varTypeIsValidHfaType(T vt)
     }
 }
 
+inline bool TypeIsInList(var_types type1, var_types type2)
+{
+    return type1 == type2;
+}
+
+// Check if type1 matches any type from the list.
+template <typename... T>
+bool TypeIsInList(var_types type1, var_types type2, T... rest)
```

Comment:
@AndyAyersMS was right in https://github.com/dotnet/runtime/pull/43386#discussion_r516890967.
I am looking for a better name for this method, could not keep `TypeIs` because that would be a conflict inside `GenTree` methods.

Reply:
Prior art here is `GenTree::StaticOperIs`, I personally do not like that name.

I would be voting for `V/varTypeIs` (both cases are in use today...).